## Import wooldridge library

In [1]:
import wooldridge as woo

## Load dataset

In [2]:
df = woo.data('bwght')
df.head()

faminc  cigtax    cigprice  bwght  fatheduc  motheduc  parity  male  white  \
0    13.5    16.5  122.300003    109      12.0      12.0       1     1      1   
1     7.5    16.5  122.300003    133       6.0      12.0       2     1      0   
2     0.5    16.5  122.300003    129       NaN      12.0       2     0      0   
3    15.5    16.5  122.300003    126      12.0      12.0       2     1      0   
4    27.5    16.5  122.300003    134      14.0      12.0       2     1      1   

   cigs    lbwght  bwghtlbs  packs   lfaminc  
0     0  4.691348    6.8125    0.0  2.602690  
1     0  4.890349    8.3125    0.0  2.014903  
2     0  4.859812    8.0625    0.0 -0.693147  
3     0  4.836282    7.8750    0.0  2.740840  
4     0  4.897840    8.3750    0.0  3.314186

## (i) How many women are in the sample and how many reported smoking during pregnancy?

In [3]:
# Count the total number of women in the sample
n_total = df.shape[0]

# Count how many women reported smoking during pregnancy (cigs > 0)
n_smokers = (df['cigs'] > 0).sum()

# Print results
print(f"Total number of women in the sample: {n_total}")
print(f"Number of women who smoked during pregnancy: {n_smokers}")

Total number of women in the sample: 1388
Number of women who smoked during pregnancy: 212


## (ii) What is the average number of cigarettes smoked per day?

In [4]:
#calculating the average number of cigarettes
mean_cigs = df['cigs'].mean()

# Display result
print(f"Average number of cigarettes consumed per day: {mean_cigs:.2f} cigarettes")

Average number of cigarettes consumed per day: 2.09 cigarettes


## (iii) Among women who smoked during pregnancy, what is the average number of cigarettes smoked per day? How does this compare with the answer to item (ii)?

In [5]:
mean_cigs_smokers = df.loc[df['cigs'] > 0, 'cigs'].mean()

print(f"Average number of cigarettes smoked by women who reported smoking during pregnancy per day: {mean_cigs_smokers:.2f}")
print(f"Difference from the overall average of 'cigs': {mean_cigs - mean_cigs_smokers:.2f}")

Average number of cigarettes smoked by women who reported smoking during pregnancy per day: 13.67
Difference from the overall average of 'cigs': -11.58


Interpretation

Overall average (ii): diluted because it includes many 0s (non-smokers).

Average among smokers (iii): much higher, since it excludes those 0s.

The comparison shows how misleading the overall mean can be if the sample has many non-smokers.

## (iv) Find the mean of 'fatheduc' in the sample. Why are only 1,192 observations used to calculate this mean?

In [6]:
# Calculate the average years of father's education (ignoring missing values by default)
mean_fatheduc = df['fatheduc'].mean()

# Count how many non-missing observations are used
n_non_missing = df['fatheduc'].count()

# Count total number of women in the sample
n_total = df.shape[0]

print(f"Average father's education (years): {mean_fatheduc:.2f}")
print(f"Number of observations used: {n_non_missing}")
print(f"Total sample size: {n_total}")
print(f"Missing values in fatheduc: {n_total - n_non_missing}")

Average father's education (years): 13.19
Number of observations used: 1192
Total sample size: 1388
Missing values in fatheduc: 196


The .mean() function automatically excludes missing values (NaN).

That’s why the average is calculated using 1,192 observations, not the full sample (≈1,388).

The difference comes from mothers who did not report the father’s education, leaving missing values in fatheduc.